### DISCUSSION TITLE
How do i execute my repo?

### DISCUSSION THREAD

**Discussion by IsaqueCar** [community/community#138699] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I wan't to test an OS i'm making here on Github, since i already added the bootloader i need a safe VCPU to work with.
Is it possibleto execute assembly here and are the devtools good?

**Comment by ferid333** [comment]
Hello @IsaqueCar !

GitHub itself doesn’t provide a direct way to execute code, especially something low-level like assembly, or run a virtual CPU (VCPU) for testing an operating system. However, you can still use GitHub for your OS project with these approaches:

1. **Use GitHub Actions for Automation:**  
   You can automate tests or builds using GitHub Actions, but it won’t give you a full environment to run an OS. You can configure Actions to build your OS, run unit tests, and automate the workflow.

2. **Use a Local Emulator (e.g., QEMU or Bochs):**  
   For testing your OS safely, you can use emulators like QEMU or Bochs locally. These provide virtual CPUs where you can run and test your OS. You can commit your code and scripts to GitHub, and then run it locally on the emulator.

3. **Use DevTools like VSCode with Extensions:**  
   To write assembly and test it, you can use development tools like Visual Studio Code (VSCode) with relevant extensions (for assembly syntax highlighting, debugging, etc.), and then commit your progress to GitHub.
